mcstatus/querier.py

```diff
@@ -170,11 +88,37 @@ def from_connection(cls, response: Connection) -> Self:
 
         response.read(len("player_") + 1 + 1)
 
-        players = []
+        players_list = []
         while True:
             player = response.read_ascii()
             if len(player) == 0:
                 break
-            players.append(player)
+            players_list.append(player)
+
+        return RawQueryResponse(**data), players_list
 
-        return cls(data, players)
+
+class AsyncServerQuerier(ServerQuerier):
```

Comment:
I'm not sure I love this inheritance. It breaks [LSP](https://en.wikipedia.org/wiki/Liskov_substitution_principle), since it inherits from a class that's synchronous, and changes the methods in it to async ones. Ideally, there'd be a private base class, and 2 classes that inherit from this class, with sync and async alternatives.

Reply:
I don't really think that refactoring of those two classes should belong to this PR.